portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,203 @@
+var GoogleAuth;
+
+// Scope for read access to Tasks API
+const SCOPE_TASKS_READ = 'https://www.googleapis.com/auth/tasks.readonly';
+
+const TASKS_CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+const TASKS_DISCOVERY_DOCS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+var TASKS_API_KEY = '';
+
+function fetchApiKey() {
+  fetch('./appConfigServlet').then(response => {
+    TASKS_API_KEY = response.json()['API_KEY'];
+  });
+}
+
+function handleClientLoadTasks() {
+  // Load the API's client and auth2 modules.
+  // Call the initClient function after the modules load.
+  gapi.load('client:auth2', initClientTasks);
+}
+
+function initClientTasks() {
+  fetchApiKey();
+
+  // Initialize the gapi.client object, which app uses to make API requests.
+  // Get API key and client ID from API Console.
+  // 'scope' field specifies space-delimited list of access scopes.
+  gapi.client
```

Comment:
I think it might be possible that when this line runs the API key is still being fetched remotely. So I think you can consider chaining this gapi call to fetchApiKey.

Reply:
Ya I think you're right James. I'd be surprised if this works as written because `TASKS_API_KEY` will almost certainly still be `null` when you make this call.

Reply:
Done.